lms/djangoapps/onboarding_survey/forms.py

```diff
@@ -47,6 +49,32 @@ def clean_country_of_employment(self):
                 )
         return self.cleaned_data['country_of_employment']
 
+    def clean_country_of_residence(self):
+        curr_dir = os.path.dirname(__file__)
+        all_countries_json_file = open(os.path.join(curr_dir, 'data/world_countries.json'))
+        all_countries = json.load(all_countries_json_file)
+
+        country_of_residence = self.cleaned_data['country_of_residence']
+
+        for country in all_countries:
+            if country == country_of_residence:
+                return self.cleaned_data['country_of_residence']
+
+        raise forms.ValidationError('Please select country of residence from the auto-fill drop down.')
```

Comment:
i think "Please select country of residence" message should be enough. No need to mention auto-fill filed